caikit_tgis_backend/managed_tgis_subprocess.py

```diff
@@ -0,0 +1,371 @@
+
+    def _monitor_bootup(self):
+        """Ran in a background thread to complete the BOOTING -> READY transition
+
+        Polls the TGIS process until if finishes booting. The polling is guarded
+        with a timeout.
+        """
+        # Wait for the server to be ready
+        start_t = time.time()
+        while True:
+            time.sleep(self._bootup_poll_delay)
+
+            # if we are not still booting, exit
+            # another thread could have caused a state transition
+            if self._tgis_state != _TGISState.BOOTING:
+                return
+
+            # check if the process stopped/crashed
+            with self._mutex:
+                try:
+                    if self._tgis_proc.poll() is not None:
+                        if self._tgis_state != _TGISState.STOPPED:
+                            self._ensure_terminated()
+                        exc = RuntimeError(
+                            "TGIS failed to boot up with the model. See logs for details"
+                        )
+                        self._bootup_exc = exc
+                        error("<MTS11752287E>", exc)
+                except AttributeError:
+                    if self._tgis_state != _TGISState.STOPPED:
+                        self._ensure_terminated()
+                    exc = RuntimeError(
+                        "TGIS process terminated while waiting for it to boot with the model"
+                    )
+                    self._bootup_exc = exc
+                    error("<MTS26557152E>", exc)
+
+            # see if we can get a passing health check request
+            try:
+                if self._health_check():
+                    with self._mutex:
+                        # double check that we are still in the booting state
+                        if not self._tgis_state == _TGISState.BOOTING:
+                            return
+                        # if the health check passes, we are good to go
+                        self._create_grpc_client()
+                        log.debug("_monitor_bootup setting state to READY")
+                        self._tgis_state = _TGISState.READY
+                        log.debug("TGIS booted for model [%s]", self._model_path)
+                        return
+            except requests.exceptions.RequestException as e:
+                # Ignore ConnectionErrors that are expected during boot up
+                if isinstance(e, requests.exceptions.ConnectionError):
+                    continue
+                log.warning(
+                    "<MTS96271549W>",
+                    "TGIS health check failed in _monitor_bootup: %s",
+                    repr(e),
+                )
+
+            # limit how long we wait before giving up
+            if time.time() - start_t >= self._load_timeout:
+                with self._mutex:
+                    # double check that we are still in the booting state
+                    if not self._tgis_state == _TGISState.BOOTING:
```

Comment:
Curious why `not X == Y` vs `X != Y`? No real preference on my end

Reply:
Writing code is sometimes a stochastic process with environmental factors, aka I also didn't have a strong preference and didn't choose consistently :P

I'll make em `X != Y`.